src/main/java/sbt_inc/SbtIncrementalCompiler.java

```diff
@@ -304,12 +316,10 @@ private File getCompiledBridgeJar(ScalaInstance scalaInstance, Log mavenLogger)
       // compile and install
       RawCompiler rawCompiler = new RawCompiler(scalaInstance, ClasspathOptionsUtil.auto(), logger);
 
-      File bridgeSources =
-          resolver.getJar(SBT_GROUP_ID, bridgeArtifactId, zincVersion, "sources").getFile();
+      File bridgeSources = resolver.getJar(groupId, bridgeArtifactId, version, "sources").getFile();
 
```

Comment:
I forgot to say that the Scala 3 compiler bridge (`scala3-sbt-bridge`) is pure java. That means you don't need to fetch the sources and to compile them. You can use the jar files directly.

So probably the code will become:

```
if (isScala3) {
  return resolver.getJar(groupId, bridgeArtifactId, version).getFile(); // fetch the jar and use as it is
} else {
  // fetch the sources
  File bridgeSources = resolver.getJar(groupId, bridgeArtifactId, version, "sources").getFile();

  // compile the sources
  // build a jar file
}

```

Reply:
> I forgot to say that the Scala 3 compiler bridge (scala3-sbt-bridge) is pure java. That means you don't need to fetch the sources and to compile them. You can use the jar files directly.

![4318779](https://user-images.githubusercontent.com/849197/113113269-682c6400-920a-11eb-9bc1-e01a77516dec.jpg)